### PR TITLE
Do::Lesson Update Lesson

### PR DESCRIPTION
**Pull request by TrucLuu**

### CODE REVIEW

File: app/controllers/lessons_controller.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    @category = Category.find params[:category_id]
+    @lesson = Lesson.find params[:id]
+  end
+
+  def index
+    @user = User.find current_user
+    @lesson = @user.lessons
+  end
+
+  def new
```
Comment (dieunb) [inline app/controllers/lessons_controller.rb]: no need

File: app/controllers/lessons_controller.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      redirect_to category_lesson_path @category, @lesson, check: 1
+    else
+      flash[:danger] = I18n.t "not-enough-word"
+      redirect_to :back
+    end
+  end
+
+  private
+  def lesson_params
+    params.require(:lesson).permit results_attributes: [:word_id, :answer_id]
```
Comment (dieunb) [inline app/controllers/lessons_controller.rb]: missing `id` result, and no need `word_id`

File: app/models/lesson.rb [hunk 3]
```diff
@@ -9,4 +15,20 @@ class Lesson < ActiveRecord::Base
   scope :lesson_learned, ->user{where("user_id = ?", user.id)}
   scope :follow_learned, ->user{where("user_id IN (
     SELECT followed_id FROM relationships WHERE follower_id = #{user.id}) OR user_id = ?", user.id)}
+
+  accepts_nested_attributes_for :results, allow_destroy: true
+
+  private
+  def add_words_to_lesson
+    Word.random(user: user).each do |word|
```
Comment (dieunb) [inline app/models/lesson.rb]: from category?

File: app/controllers/lessons_controller.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
+class LessonsController < ApplicationController
+  before_action :logged_in_user
+
+  MAX_WORD = Settings.length.words
+
+  def show
```
Comment (khanhkhanh) [inline app/controllers/lessons_controller.rb]: index => shơư

File: app/controllers/lessons_controller.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    @user = User.find current_user
+    @lesson = @user.lessons
+  end
+
+  def show
+    @lesson = Lesson.find params[:id]
+    @lesson_words = @lesson.results
+  end
+
+  def create # cai tao 20 word o dau model model n
```
Comment (dieunb) [inline app/controllers/lessons_controller.rb]: remove all remark

**Pull request merged**